src/main/kotlin/lotto/domain/Lotto.kt

```diff
@@ -1,11 +1,27 @@
 package lotto.domain
 
-class Lotto private constructor(val lottoNumbers: Set<LottoNumber>) {
-    companion object {
-        fun createLotto(numbers: List<Int>): Lotto {
-            val lottoNumbers = numbers.map { LottoNumber.from(it) }.toSet()
+import lotto.constant.REQUIRED_LOTTO_SIZE
+
+class Lotto(val lottoNumbers: Set<LottoNumber>) {
+    constructor(vararg numbers: Int) : this(numbers.map(::LottoNumber).toSet())
 
-            return Lotto(lottoNumbers)
+    init {
+        require(lottoNumbers.size == REQUIRED_LOTTO_SIZE) {
+            ERROR_WRONG_NUMBER_COUNT
         }
     }
+
+    fun getIntersectSize(winningLotto: Lotto): Int {
+        val lottoNumbers: Set<LottoNumber> = lottoNumbers
+        val winningLottoNumbers: Set<LottoNumber> = winningLotto.lottoNumbers
```

Comment:
```suggestion
        val lottoNumbers = lottoNumbers
        val winningLottoNumbers = winningLotto.lottoNumbers
```

타입에 대한 부분은 생략해줘도 좋아보여요~ 코틀린의 장점은 타입 추론이 가능하다는 부분이에요. 타입 추론이 있기에 더욱더 명확한 변수 명이 나온다고 생각해요. 명확한 변수 명이 나오면 타입을 어느정도 추론할 수 있다고 봅니다. 그리고 IDE에서 타입을 다 알려주잖아요? ㅎㅎ